packages/medusa/src/models/batch-job.ts

```diff
@@ -159,43 +177,47 @@ export class BatchJob extends SoftDeletableEntity {
  *          - variant.prices
  *          - images
  *  dry_run:
+ *    description: Specify if the job must apply the modifications or not.
  *    type: boolean
- *    description: "Specify if the job must apply the modifications or not."
  *    default: false
  *  result:
- *    type: object
- *    description: "The result of the batch job."
- *    properties:
- *      count:
- *        type: number
- *      advancement_count:
- *        type: number
- *      progress:
- *        type: number
- *      errors:
- *        type: object
- *        properties:
- *          message:
- *            type: string
- *          code:
- *            oneOf:
- *              - type: string
- *              - type: number
- *          err:
- *            type: array
- *      stat_descriptors:
- *        type: object
- *        properties:
- *          key:
- *            type: string
- *          name:
- *            type: string
- *          message:
- *            type: string
- *      file_key:
- *        type: string
- *      file_size:
- *        type: number
+ *    description: The result of the batch job.
+ *    nullable: true
+ *    allOf:
+ *    - type: object
+ *      example: {}
+ *    - type: object
+ *      properties:
+ *        count:
+ *          type: number
+ *        advancement_count:
+ *          type: number
+ *        progress:
+ *          type: number
+ *        errors:
+ *          type: object
+ *          properties:
+ *            message:
+ *              type: string
+ *            code:
+ *              oneOf:
+ *                - type: string
+ *                - type: number
+ *            err:
+ *              type: array
+ *        stat_descriptors:
+ *          type: object
+ *          properties:
+ *            key:
+ *              type: string
+ *            name:
+ *              type: string
+ *            message:
+ *              type: string
+ *        file_key:
+ *          type: string
+ *        file_size:
+ *          type: number
```

Comment:
Since it represents a JSON stored in the DB and it can take any shape, I didn't come up with a specific convention for the structure of their content. I decided to simply match the TS typing already in place. I would recommend that we keep it as is.